### PR TITLE
Create pythonpublish.yml

### PR DESCRIPTION
**Pull request by dciborow**
### Description

Adding Github Action to upload to PyPii

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: .github/workflows/pythonpublish.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Upload Python Package
```
Comment (miguelgfierro) [inline .github/workflows/pythonpublish.yml:1]: we need to take a look at how to do this, probably discuss how we want to publish the three environments (cpu, gpu and spark)

File: .github/workflows/pythonpublish.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+      run: |
+        python setup.py sdist bdist_wheel
```
Comment (gramhagen) [inline .github/workflows/pythonpublish.yml:25]: i'm assuming this build a package that works for any python3 version?

File: .github/workflows/pythonpublish.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        twine upload dist/*      
+    - name: Update version for next release
```
Comment (gramhagen) [inline .github/workflows/pythonpublish.yml:27]: we need to sync on version numbering too, we're currently using calendar versioning not semantic

**Review by miguelgfierro — APPROVED** [review]

File: .github/workflows/pythonpublish.yml (hunk 3 above)
Comment (dciborow) [reply .github/workflows/pythonpublish.yml:27]: i have very little memory writing this, but what if we stick your the calendar versioning, with an auto generated patch number if we release more then once in a month?
2021.2.0
2021.2.1
2021.3.0
?

Comment (gramhagen) [reply .github/workflows/pythonpublish.yml:27]: works for me, surely github actions have a version incrementing scheme?

**Pull request closed without merging**